api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -275,6 +275,31 @@ message ExternalProcessor {
   // backend stream lifetime. In this case, Envoy will eventually timeout the external processor stream according to this time limit.
   // The default value is 5000 milliseconds (5 seconds) if not specified.
   google.protobuf.Duration deferred_close_timeout = 19;
+
+  // [#not-implemented-hide:]
+  // Enables sending body to the side stream server when the header external processing is still ongoing.
+  // It only works for STREAMED body processing mode. For any other body processing modes, it is ignored.
+  //
+  // In the use cases if the side stream server must wait for at least one chunk of body before sending back
+  // the response for the headers, this filter configuration has to be enabled to support this.
+  //
+  // The side stream server must not send a separate header response in this case. Instead, it encodes the
+  // response headers in the first body response and sends back to Envoy.
```

Comment:
Not sure if I fully grok this, but it seems to me that the side-stream needs to know whether the server's `send_body_before_recv_header_resp` config knob is set or not.
If so, IMHO this is not a good design, as it couples the config that Envoy receives with the side-stream behavior. This may interfere with gradual rollout of a new config, or changing a service from one type to another.
It would be better to allow the side-stream send a header-response even if this field is set to true, or to "negotiate" that behavior (Envoy sending that it is going to use that behavior) to the side-stream.

Reply:
Thanks for the comments!

This configuration is very much like the processing_mode API: https://github.com/envoyproxy/envoy/blob/d78e8138feb499147730a1c39da8e479a796a621/api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto#L153.  Customer has to write their side stream server corresponding to this configuration. So if customer configured this, that means the side stream server is working in this mode. So, communicating this to the sidestream server seems to be redundant to me.  BTW, we do not communicate processing_mode configuration to the side stream server either.

Reply:
BTW, the goal of this API is to relax this condition for certain side stream servers:
https://github.com/envoyproxy/envoy/blob/d78e8138feb499147730a1c39da8e479a796a621/source/extensions/filters/http/ext_proc/ext_proc.cc#L495.
So, the body can be sent to the side stream server even the header response is not received yet.
The use case is that certain side stream server has to to see body before they can send back the response for headers: https://github.com/envoyproxy/envoy/issues/30810..

Reply:
Regarding whether Envoy can use handshake with the side stream server to "negotiate" the behavior, we had discussed this offline. The answer is no.  The reason is that headers and bodies are flowing into Envoy from downstream client, and Envoy suppose to send them to the side stream server right away. There is no time for Envoy to wait for a response which carry that info.

Reply:
It looks to me adding a field in the ProcessRequest API to notify this Envoy filter configuration is cleaner.  Just updated the change.